protocol/invocation.go

```diff
@@ -57,4 +58,6 @@ type Invocation interface {
 	SetAttribute(key string, value interface{})
 	GetAttribute(key string) (interface{}, bool)
 	GetAttributeWithDefaultValue(key string, defaultValue interface{}) interface{}
+
+	ToContext() context.Context
```

Comment:
GetContext() may be better?

Reply:
I changed to 'GetAttachmentAsContext', which is more clear.